dartagnan/src/main/java/com/dat3m/dartagnan/encoding/ProgramEncoder.java

```diff
@@ -545,99 +536,119 @@ private BooleanFormula encodeFairForwardProgress(Thread thread) {
             return bmgr.and(enc);
         }
 
-        /*
-            Minimal progress means that at least one thread must get scheduled, i.e., the execution cannot just stop.
-            In particular, a single-threaded program must progress and concurrent programs with only finite threads
-            must terminate (unless blocked).
-
-            NOTE: For producing correct verdicts on loop-based nontermination, we do not really require
-                  minimal progress guarantees.
-                  This may change in the presence of control barriers, depending on how they affect scheduling,
-                  i.e., do blocked threads remain eligible for scheduling?
-         */
-        private BooleanFormula encodeMinimalProgress(Program program) {
-            final BooleanFormulaManager bmgr = context.getBooleanFormulaManager();
+        // ================================================= New Encoding =================================================
```

Comment:
Why "New"? Are you still keeping the legacy one? If this one is working, we should get rid of the other one